pkg/kubectl/resource_printer.go

```diff
@@ -350,106 +349,61 @@ func podHostString(host, ip string) string {
-func interpretContainerStatus(status *api.ContainerStatus) (string, string, string, error) {
-	// Helper function to compose a meaning message from terminate state.
-	getTermMsg := func(state *api.ContainerStateTerminated) string {
-		var message string
-		if state != nil {
-			message = fmt.Sprintf("exit code %d", state.ExitCode)
-			if state.Reason != "" {
-				message = fmt.Sprintf("%s, reason: %s", message, state.Reason)
-			}
-		}
-		return message
-	}
-
-	state := &status.State
-	if state.Waiting != nil {
-		return "Waiting", "", state.Waiting.Reason, nil
-	} else if state.Running != nil {
-		// Get the information of the last termination state. This is useful if
-		// a container is stuck in a crash loop.
-		message := getTermMsg(status.LastTerminationState.Terminated)
-		if message != "" {
-			message = "last termination: " + message
-		}
-		stateMsg := "Running"
-		if !status.Ready {
-			stateMsg = stateMsg + " *not ready*"
-		}
-		return stateMsg, translateTimestamp(state.Running.StartedAt), message, nil
-	} else if state.Terminated != nil {
-		return "Terminated", translateTimestamp(state.Terminated.StartedAt), getTermMsg(state.Terminated), nil
-	}
-	return "", "", "", fmt.Errorf("unknown container state %#v", *state)
+	return shortHumanDuration(time.Now().Sub(timestamp.Time))
 }
 
 func printPod(pod *api.Pod, w io.Writer, withNamespace bool) error {
-	var name string
+	name := pod.Name
 	if withNamespace {
-		name = types.NamespacedName{pod.Namespace, pod.Name}.String()
-	} else {
-		name = pod.Name
+		name = types.NamespacedName{Namespace: pod.Namespace, Name: pod.Name}.String()
 	}
 
-	_, err := fmt.Fprintf(w, "%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\n",
-		name,
-		pod.Status.PodIP,
-		"", "",
-		podHostString(pod.Spec.NodeName, pod.Status.HostIP),
-		formatLabels(pod.Labels),
-		pod.Status.Phase,
-		translateTimestamp(pod.CreationTimestamp),
-		pod.Status.Message,
-	)
-	if err != nil {
-		return err
+	restarts := 0
+	totalContainers := len(pod.Spec.Containers)
+	readyContainers := 0
+	reason := string(pod.Status.Phase)
+
+	// TODO: replace pod.Status.Message with pod.Status.Reason (to be added soon)
+	if pod.Status.Message != "" {
+		reason = pod.Status.Message
 	}
-	// Lay out all containers on separate lines.
-	statuses := pod.Status.ContainerStatuses
-	if len(statuses) == 0 {
-		// Container status has not been reported yet. Print basic information
-		// of the containers and exit the function.
-		for _, container := range pod.Spec.Containers {
-			_, err := fmt.Fprintf(w, "%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\n",
-				"", "", container.Name, container.Image, "", "", "", "", "")
-			if err != nil {
-				return err
-			}
+
+	for _, container := range pod.Status.ContainerStatuses {
+		restarts += container.RestartCount
+		if container.Ready {
+			readyContainers++
+		} else if container.State.Terminated != nil && container.State.Terminated.Reason != "" {
+			reason = container.State.Terminated.Reason
```

Comment:
I'd pick the first one instead, since users will probably list containers in order of importance.

And check Terminated before Ready, as per the comment above.